eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/ChannelOpenSingleFunder.scala

```diff
@@ -316,7 +323,20 @@ trait ChannelOpenSingleFunder extends FundingHandlers with ErrorHandlers {
             }
           }
 
-          goto(WAIT_FOR_FUNDING_CONFIRMED) using DATA_WAIT_FOR_FUNDING_CONFIRMED(commitments, Some(fundingTx), blockHeight, None, Left(fundingCreated)) storing() calling publishFundingTx()
+          Funding.minDepthFunder(commitments.channelFeatures) match {
+            case Some(fundingMinDepth) =>
+              blockchain ! WatchFundingConfirmed(self, commitInput.outPoint.txid, fundingMinDepth)
+              log.info(s"committing txid=${fundingTx.txid}")
+              goto(WAIT_FOR_FUNDING_CONFIRMED) using DATA_WAIT_FOR_FUNDING_CONFIRMED(commitments, Some(fundingTx), blockHeight, None, Left(fundingCreated)) storing() calling publishFundingTx()
+            case None =>
+              blockchain ! WatchFundingLost(self, commitments.commitInput.outPoint.txid, nodeParams.channelConf.minDepthBlocks)
+              val channelKeyPath = keyManager.keyPath(commitments.localParams, commitments.channelConfig)
+              val nextPerCommitmentPoint = keyManager.commitmentPoint(channelKeyPath, 1)
+              val shortIds = ShortIds(RealScidStatus.Unknown, ShortChannelId.generateLocalAlias(), None)
```

Comment:
This would create an issue when migrating old versions of `DATA_WAIT_FOR_FUNDING_CONFIRMED`, because we don't have an scid there to use in the alias field and we're not sure the alias generation function will be suitable to call during DB migration.

I instead refactored that code to a single function that is called in all three places where we do 0-conf in https://github.com/ACINQ/eclair/pull/2310/commits/fcb0fe4bfdb47e5e86defee33ef258cfd3e2e414

Reply:
Went a bit further with in branch [zeroconf-scid-alias-bast-factor](https://github.com/ACINQ/eclair/tree/zeroconf-scid-alias-bast-factor) and factored in the 3rd transition. The semantics of `skipFundingConfirmation` changes to `acceptFundingTx`.

In a second commit I removed the "optimization" that makes us eagerly process remote's alias to emit less events. I think it's not worth it considering how it complicates the method signature.